llvm/include/llvm/IR/IntrinsicsDirectX.td

```diff
@@ -90,4 +90,5 @@ def int_dx_rsqrt  : DefaultAttrsIntrinsic<[llvm_anyfloat_ty], [LLVMMatchType<0>]
 def int_dx_wave_is_first_lane : DefaultAttrsIntrinsic<[llvm_i1_ty], [], [IntrConvergent]>;
 def int_dx_sign : DefaultAttrsIntrinsic<[LLVMScalarOrSameVectorWidth<0, llvm_i32_ty>], [llvm_any_ty]>;
 def int_dx_step : DefaultAttrsIntrinsic<[LLVMMatchType<0>], [llvm_anyfloat_ty, LLVMMatchType<0>]>;
+def int_dx_fmod : DefaultAttrsIntrinsic<[LLVMMatchType<0>], [llvm_anyfloat_ty, LLVMMatchType<0>]>;
```

Comment:
This isn't how i invisioned this see this: https://github.com/llvm/llvm-project/issues/99118#issuecomment-2294511442

We should use FREM
https://github.com/llvm/llvm-project/blob/f564a48f0ea4d2100c0cadfa6e6f20f97244025e/llvm/include/llvm/IR/IRBuilder.h#L1667-L1676

Reply:
Unlike SPIRV backend, currently the Directx backend only lowers the intrinsics not instructions. I have a follow-up PR will expand this dx specific intrinsic.

Reply:
why can't you use `Intrinsic::experimental_constrained_frem`?

Reply:
we can modify the intrinsic expansion pass to catch frem. there is no reason that pas just has to work on intrinsics.

Reply:
Done.